auth/src/main/java/com/firebase/ui/auth/ui/email/CheckEmailFragment.java

```diff
@@ -94,6 +97,26 @@ public static CheckEmailFragment getInstance(@NonNull FlowParameters flowParamet
         return fragment;
     }
 
+    public static Task<String> fetchTopProvider(FirebaseAuth auth, @NonNull String email) {
+        if (TextUtils.isEmpty(email)) {
+            return Tasks.forException(new IllegalArgumentException("Email cannot be empty"));
+        }
+
+        return auth.fetchProvidersForEmail(email)
+                .addOnFailureListener(
+                        new TaskFailureLogger(TAG, "Error fetching providers for email"))
+                .continueWith(new Continuation<ProviderQueryResult, String>() {
+                    @Override
+                    public String then(@NonNull Task<ProviderQueryResult> task) throws Exception {
+                        if (!task.isSuccessful()) return null;
+
+                        List<String> providers = task.getResult().getProviders();
+                        if (providers == null || providers.isEmpty()) return null;
```

Comment:
Maybe I am dumb but why switch from `.get(0)` to `.get(size() - 1)` ?
 
Also style nit but for blocks with `if` and `else` can we use `{}` even when each clause is one line?